google_cloud_http_target_task.js

```diff
@@ -0,0 +1,63 @@
+const {CloudTasksClient} = require('@google-cloud/tasks');
+const yenv = require('yenv')
+const { v4: uuidv4 } = require('uuid');
+const fs = require('fs')
+const os = require('os');
+
+// Parse 'gql-perf-arch-github_workflow.yml' file
+const parsedDocument = yenv('.github/workflows/gql-perf-arch-github_workflow.yml')
+
+// Instantiate a client
+const client = new CloudTasksClient();
+
+// Payload variables
+const jobId = String(uuidv4());
+const commitHash = parsedDocument.COMMIT_HASH;
+const classes = parsedDocument.CLASSES;
+const pullRequestNumber = parsedDocument.PULL_REQUEST_NUMBER;
+
+
+async function createHttpTaskWithToken() {
+  // Actions secrets (keys)
+  const project = parsedDocument.PROJECT_ID;
+  const queue = parsedDocument.QUEUE_ID;
+  const location = parsedDocument.LOCATION;
+  const url = parsedDocument.WORKFLOW_URL;
+  const serviceAccountEmail = parsedDocument.SERVICE_ACCOUNT_EMAIL;
+
+  // Construct payload
+  const payloadStructure =  {
+                                 "jobId": jobId,
+                                 "commitHash": commitHash,
+                                 "classes": classes,
+                                 "pullRequest": pullRequestNumber,
+                            }
+  //Format payload
```

Comment:
I prefer to have comments only for un-intuitive bits of code. Where you have a straight forward comment like this, I think really what you're after is a helper method definition. For example

const formatPayload = () => {
   // whatever you need to do to format the payload.
}

Reply:
Done, I tried to reduce the work load of the main function by creating multiple self documented functions.